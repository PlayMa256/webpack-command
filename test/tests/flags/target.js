const { apply, build, crcDist, test, validate } = require('../../util');

const fixture = 'target/target';
const opts = { fixture };

let config;

test('--target', module, () => {
  it(`should validate`, () => {
    expect(validate(opts)).toEqual(true);
  });

  it(`should apply`, () => {
    config = apply(opts);

    expect(config).toMatchSnapshot();
  });

  it(`should build`, () =>
    build(config).then((result) => {
      expect(result).toMatchSnapshot();
      expect(crcDist()).toMatchSnapshot();
    }));
});
