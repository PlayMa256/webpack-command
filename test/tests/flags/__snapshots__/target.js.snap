// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --target > should apply #0 1`] = `
Object {
  "devtool": "source-map",
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
  "stats": Object {
    "builtAt": false,
    "hash": false,
    "timings": false,
  },
  "target": "node",
}
`;

exports[`Flags > --target > should build #0 1`] = `
"Version: webpack 4.3.0
      Asset      Size  Chunks             Chunk Names
    main.js  3.13 KiB    main  [emitted]  main
main.js.map  2.85 KiB    main  [emitted]  main
Entrypoint main = main.js main.js.map
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --target > should build #1 1`] = `"c4994062"`;
